src/colmap/geometry/rigid3.h

```diff
@@ -54,6 +54,20 @@ struct Rigid3d {
     matrix.col(3) = translation;
     return matrix;
   }
+
+  // Adjoint matrix to propagate uncertainty on Rigid3d
+  // [Reference] https://gtsam.org/2021/02/23/uncertainties-part3.html
+  inline Eigen::Matrix6d Adjoint() const {
+    Eigen::Matrix6d adjoint = Eigen::Matrix6d::Zero();
+    Eigen::Matrix3d skew_t;
+    skew_t << 0, -translation(2), translation(1), translation(2), 0,
+        -translation(0), -translation(1), translation(0), 0;
+    Eigen::Matrix3d rot_matrix = rotation.toRotationMatrix();
+    adjoint.block<3, 3>(0, 0) = rot_matrix;
+    adjoint.block<3, 3>(3, 0) = skew_t * rot_matrix;
+    adjoint.block<3, 3>(3, 3) = rot_matrix;
```

Comment:
```suggestion
    Eigen::Matrix6d adjoint;
    Eigen::Matrix3d skew_t;
    skew_t << 0, -translation(2), translation(1), translation(2), 0,
        -translation(0), -translation(1), translation(0), 0;
    adjoint.block<3, 3>(0, 0) = rotation.toRotationMatrix();
    adjoint.block<3, 3>(0, 3).setZero();
    adjoint.block<3, 3>(3, 0) = translation.colwise().cross(adjoint.block<3, 3>(0, 0));
    adjoint.block<3, 3>(3, 3) = adjoint.block<3, 3>(0, 0);
```

Reply:
Thanks! However, there is slight typo here so this wont compile since colwise() is applied at the wrong place. Moreover, translation.cross(A.colwise()) is not supported in Eigen, so I updated it to A.colwise().cross(-translation).